File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/data/type/StringValueType.ts
module api.data.type {

    export class StringValueType extends ValueType {

        constructor() {
            super("String");
        }

        isValid(value: any): boolean {
            return typeof value === 'string';
        }

        valueEquals(a: string, b: string): boolean {
            return api.ObjectHelper.stringEquals(a, b);
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/data/StringValueType.java
package com.enonic.xp.data;

final class StringValueType
    extends ValueType<String>
{
    StringValueType()
    {
        super( "String", JavaTypeConverters.STRING );
    }

    @Override
    public Value fromJsonValue( final Object object )
    {
        return ValueFactory.newString( convertNullSafe( object ) );
    }
}
